components/bl-leafletMap-component/src/helpers/map.js

```diff
@@ -0,0 +1,173 @@
+import Leaflet from '../lib/leaflet/leaflet';
+import { toCoordinates } from './coordinates';
+import { Maps } from '../maps';
+
+export function changeMapType(map, currentLayer, component) {
+  const { mapType } = component;
+
+  if (mapType === undefined) {
+    return;
+  } else if (mapType in Maps) {
+    map.current.removeLayer(currentLayer.current);
+
+    currentLayer.current = Leaflet.tileLayer(Maps[mapType].mapUrl, Maps[mapType].options).addTo(map.current);
+  } else {
+    console.error('Leaflet Map: not valid map type');
+  }
+}
+
+export function getGeolocation(map, geoMarker, icon, eventHandlers) {
+  const { onDeterminingGeoposition } = eventHandlers;
+
+  navigator.geolocation.getCurrentPosition(
+    pos => {
+      const coords = [pos.coords.latitude, pos.coords.longitude];
+
+      onDeterminingGeoposition({ coordinates: coords });
+
+      if (geoMarker) {
+        map.removeLayer(geoMarker);
+      }
+
+      map.setView(coords, 14);
+
+      geoMarker = Leaflet.marker(coords, { icon }).addTo(map);
```

Comment:
`geoMarker = ` is it needed?

Reply:
When we create a geolocation marker, we write it down so that the next time we find a geolocation, we delete the old marker and add a new one.

Reply:
hold on, I do not see where it's used right after the line

if the idea was to change the `geoMarker ` somewhere outside the function it won't work this way

Reply:
if (geoMarker) {
        map.removeLayer(geoMarker);
      }

      map.setView(coords, 14);

      geoMarker = Leaflet.marker(coords, { icon }).addTo(map);
      
      We check if there is currently a marker, and if it is, we delete it, then add new ones with exact coordinates.

Reply:
https://github.com/Backendless/ui-builder-library/pull/11#discussion_r923318548
If the problem was that Ref didn't have current, I solved it.